src/accelerate/accelerator.py

```diff
@@ -2095,14 +2097,14 @@ def gather(self, tensor):
         """
         return gather(tensor)
 
-    def gather_for_metrics(self, tensor):
+    def gather_for_metrics(self, input_data):
         """
-        Gathers `tensor` and potentially drops duplicates in the last batch if on a distributed system. Should be used
-        for gathering the inputs and targets for metric calculation.
+        Gathers `tensor` or object and potentially drops duplicates in the last batch if on a distributed system. For
+        objects, duplicates are removed. Should be used for gathering the inputs and targets for metric calculation.
```

Comment:
```suggestion
        Gathers `tensor` or object and potentially drops duplicates in the last batch if on a distributed system. Should be used for gathering the inputs and targets for metric calculation.
```
Since the end duplicates are removed in general, don't think we need this extra sentence.